### PR TITLE
Preparation for empirically testing shooter

### PR DESCRIPTION
**Pull request by Sajo8**
- don't do the elevation constantly (only when pressed)
- make velocitypoint class
- some initial code to return the closest value 
-> we will want to later interpolate stuff between the thing in the table and our tihng
- move shooter velocity calculation to the shooter subsystem

### CODE REVIEW

File: InfiniteRecharge/src/main/java/frc/robot/subsystem/vision/VelocityPoint.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package frc.robot.subsystem.vision;
+
+public class VelocityPoint implements Comparable<VelocityPoint> {
```
Comment (cpostbitbuckets) [inline InfiniteRecharge/src/main/java/frc/robot/subsystem/vision/VelocityPoint.java]: Just imagine how clean this class would be if everything were public variables...

File: InfiniteRecharge/src/main/java/frc/robot/subsystem/scoring/shooter/ShooterSubsystem.java [hunk 2]
```diff
@@ -336,21 +342,46 @@ public void calculateAbsoluteDegreesToRotate() {
...
+            double ty = visionSubsystem.getTy();
+            // double degrees = getTargetElevationDegGivenOffset(ty);
+            double distance = visionSubsystem.approximateDistanceFromTarget(ty);
+
+            List<VelocityPoint> velocityPoints = new ArrayList<VelocityPoint>();
+            // add values here
+ 
+            int closestIndex = Collections.binarySearch(velocityPoints, new VelocityPoint(distance, 0, 0));
+            // TODO: interpolate between closestIndex and our thing
+            return velocityPoints.get(closestIndex).getElevationAngle_deg();
```
Comment (cpostbitbuckets) [inline InfiniteRecharge/src/main/java/frc/robot/subsystem/scoring/shooter/ShooterSubsystem.java]: You'll want some error checking on this. Make sure closestIndex is actually a valid index in the list.

File: InfiniteRecharge/src/main/java/frc/robot/subsystem/scoring/shooter/ShooterSubsystem.java [hunk 3]
```diff
@@ -336,21 +342,46 @@ public void calculateAbsoluteDegreesToRotate() {
...
+        // decreases as u get closer
+        // increases as u get it farther
+    public double calculateAbsoluteDegreesElevation() {
+        boolean validTarget = visionSubsystem.getValidTarget();
+        if (validTarget) {
+            double ty = visionSubsystem.getTy();
+            // double degrees = getTargetElevationDegGivenOffset(ty);
+            double distance = visionSubsystem.approximateDistanceFromTarget(ty);
+
+            List<VelocityPoint> velocityPoints = new ArrayList<VelocityPoint>();
```
Comment (cpostbitbuckets) [inline InfiniteRecharge/src/main/java/frc/robot/subsystem/scoring/shooter/ShooterSubsystem.java]: This should be a member of the class, not a local variable.

**Comment by cpostbitbuckets** [comment]
We also want to make sure we aren't impacting our current manual control of the elevation motor.

**Comment by Sajo8** [comment]
Ok I'll just keep it on my branch for now till we get all the values in

**Pull request closed without merging**